ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -26,11 +28,15 @@
 
 using namespace llvm;
 using namespace llvm::symbolize;
+using namespace std;
 
-#define TRACE_FUNC_NAME "__yk_compiled_trace"
+#define TRACE_FUNC_PREFIX "__yk_compiled_trace_"
 #define YKTRACE_START "__yktrace_start_tracing"
 #define YKTRACE_STOP "__yktrace_stop_tracing"
 
+// An atomic counter used to issue compiled traces with unique names.
+atomic<uint64_t> NextTraceIdx(0);
```

Comment:
Instead of using a counter (when one always has to worry "what happens if this process runs for 70 years and the integer wraps), could we use the address of the trace in memory? That's guaranteed to be stable.

Reply:
Good idea.

Since raising the PR, I'm also wondering if there is any need to have a unique name except for when debugging (when you need to set a breakpoint on a specific trace -- lukas and I realised that LLVM cleverly inserts a symbol to help with this). We might choose to only name things uniquely for debug builds...

Thoughts?

Reply:
> We might choose to only name things uniquely for debug builds

Unless that causes symbol clashing later, that is. At present I'm not sure. We will find out soon.

Reply:
I have no opinion on whether we need unique names or not. That's up to you!

Reply:
I may also shorted `__yk_compiled_trace_$ptr` to `__yk_jitted_$ptr`, where `$ptr` is the hex address of the pointer (as you suggested). OK with that?

Reply:
Hold on, I don't know the address of the trace until codegen is done, at which point I'm unable to name the trace. We can allocate a byte just for the purpose of having a unique ID perhaps?

Reply:
Also found this: https://llvm.discourse.group/t/duplicate-symbols-in-separate-modules-no-longer-allowed/1805

So naming our functions uniquely is probably a good idea.

Reply:
If we don't have a simple memory address, then let's just use a counter and add something like:

```
let new_counter = old_counter.fetch_add(1);
if new_counter < old_counter { todo!(); }
```

so we catch the "process has been running for 70 years" case. It's a good habit :)

Reply:
We have a similar check [here](https://github.com/ykjit/yk/pull/349/files#diff-28de043b71aaedbd9af7218078abd71f6216cff0ce9dbb53e6518efc62817021R114).

Works for you?

Reply:
[I think your solution allows one more trace than mine!]

Reply:
Erp, that's a good point. I think the current check you have is correct (and mine is wrong).